src/bin/ion/commands/jq.rs

```diff
@@ -257,40 +291,227 @@ impl Add for JaqElement {
+
+            // Number types, only lossless operations
+            (Int(a), Int(b)) => (a.expect_i128().unwrap() / b.expect_i128().unwrap()).into(),
+            (Float(a), Float(b)) => (a / b).into(),
+            (Decimal(a), Decimal(b)) => a.div(b).into(),
+            (Decimal(a), Int(b)) => a.div(b).into(),
+            (Int(a), Decimal(b)) => a.div(b).into(),
+
+            // Only try potentially lossy Float conversions when we've run out of the other options
+            (a @ Int(_) | a @ Decimal(_), Float(b)) => {
+                let Some(f) = a.clone().to_f64() else {
+                    return jaq_unary_error(a, "cannot be an f64");
+                };
+                (f / b).into()
+            }
+            (Float(a), b @ Int(_) | b @ Decimal(_)) => {
+                let Some(f) = b.clone().to_f64() else {
+                    return jaq_unary_error(b, "cannot be an f64");
+                };
+                (a / f).into()
+            }
+
+            (a, b) => return jaq_binary_error(a, b, "cannot be divided"),
+        };
+
+        Ok(JaqElement::from(elt))
     }
 }
 
 impl Rem for JaqElement {
     type Output = ValR<Self>;
 
     fn rem(self, _rhs: Self) -> Self::Output {
-        todo!()
+        let (lhv, rhv) = (self.into_value(), _rhs.into_value());
+
+        use ion_math::{DecimalMath, ToFloat};
+        use Value::*;
+
+        let elt: Element = match (lhv, rhv) {
+            // Number types, only lossless operations
+            (Int(a), Int(b)) => (a.expect_i128().unwrap() % b.expect_i128().unwrap()).into(),
+            (Float(a), Float(b)) => (a % b).into(),
+            (Decimal(a), Decimal(b)) => a.rem(b).into(),
+            (Decimal(a), Int(b)) => a.rem(b).into(),
+            (Int(a), Decimal(b)) => a.rem(b).into(),
+
+            // Only try potentially lossy Float conversions when we've run out of the other options
+            (a @ Int(_) | a @ Decimal(_), Float(b)) => {
+                let Some(f) = a.clone().to_f64() else {
+                    return jaq_unary_error(a, "cannot be an f64");
+                };
+                (f % b).into()
+            }
+            (Float(a), b @ Int(_) | b @ Decimal(_)) => {
+                let Some(f) = b.clone().to_f64() else {
+                    return jaq_unary_error(b, "cannot be an f64");
+                };
+                (a % f).into()
+            }
+
+            (a, b) => return jaq_binary_error(a, b, "cannot be divided (remainder)"),
+        };
+
+        Ok(JaqElement::from(elt))
     }
 }
 
 impl Neg for JaqElement {
     type Output = ValR<Self>;
 
     fn neg(self) -> Self::Output {
-        todo!()
+        let val = self.into_value();
+
+        use ion_math::DecimalMath;
+        use Value::*;
+
+        let elt: Element = match val {
+            // Only number types can be negated
+            Int(a) => (-a).into(),
+            Float(a) => (-a).into(),
+            Decimal(a) => (-a.to_big_decimal()).to_decimal().into(),
```

Comment:
This works fine, but you could probably achieve the same thing much more cheaply by manually working with the `Decimal`'s coefficient and `Sign`. Definitely not a priority.